### PR TITLE
elvish 0.10.1

### PR DESCRIPTION
**Pull request by zhsj**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Elvish now supports -version and accepts -X main.Version

### CODE REVIEW

**Comment by zhsj** [comment]
sorry, we need to bump the version first.. since 0.10 is a bad tag..

**Comment by zhsj** [comment]
This replace #18231

File: Formula/elvish.rb [hunk 1]
```diff
@@ -1,8 +1,10 @@
 class Elvish < Formula
   desc "Novel UNIX shell written in Go"
   homepage "https://github.com/elves/elvish"
-  url "https://github.com/elves/elvish/archive/0.10.tar.gz"
-  sha256 "1b053abd3f8c3b436712597e5dc7ce44c71737bdc0016cbf773a4e74c712851a"
+  url "https://github.com/elves/elvish.git",
+      :tag => "0.10.1",
```
Comment (ilovezfs) [inline Formula/elvish.rb]: Why this change? If building from the archive still works, we should not switch.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/elvish.rb (hunk 1 above)
Comment (zhsj) [reply Formula/elvish.rb]: I thought this style is more preferred.. ok to revert

Comment (ilovezfs) [reply Formula/elvish.rb]: In order of preference:
1. upstream release tarball
2. source archive tarball
3. tag + revision

Comment (zhsj) [reply Formula/elvish.rb]: fixed

**Comment by ilovezfs** [comment]
Thanks @zhsj. Please squash this down to one commit with the commit subject "elvish 0.10.1"

**Review by ilovezfs — APPROVED** [review]

**Comment by zhsj** [comment]
I thought these are two changes, but squashed anyway.

**Comment by ilovezfs** [comment]
We almost always do 1 commit per formula per PR with some very occasional exceptions.

**Comment by ilovezfs** [comment]
Also 1 formula per commit :)

**Comment by ilovezfs** [comment]
Thanks @zhsj!
